### PR TITLE
VM code generation

### PR DESCRIPTION
**Pull request by mmoskal**
This implements codegen for VM running on AVR.

It additionally adds support for some recent codal build changes.

### CODE REVIEW

**Comment by mmoskal** [comment]
Tested on pxt-adafruit - no effect on ARM code

File: pxtcompiler/emitter/emitter.ts [hunk 1]
```diff
@@ -2155,7 +2165,7 @@ ${lbl}: .short 0xffff
             }
 
             // otherwise we assume a lambda
-            if (args.length > 3)
+            if (args.length > (isStackMachine() ? 2 : 3))
                 userError(9217, lf("lambda functions with more than 3 arguments not supported"))
```
Comment (pelikhan) [inline pxtcompiler/emitter/emitter.ts]: update error message with number of args?

File: pxtcompiler/emitter/hexfile.ts [hunk 2]
```diff
@@ -646,15 +646,17 @@ ${hex.hexPrelude()}
     }
 
     function mkProcessorFile(nativeType: string) {
-        let processor: assembler.AbstractProcessor = null
+        let b: assembler.File
+
         if (nativeType == NATIVE_TYPE_AVR)
```
Comment (pelikhan) [inline pxtcompiler/emitter/hexfile.ts]: use isAvr() ?

File: pxtcompiler/emitter/emitter.ts (hunk 1 above)
Comment (mmoskal) [reply pxtcompiler/emitter/emitter.ts]: didn't want to mess with localization, also I guess we may want to lift this restriction

File: pxtcompiler/emitter/hexfile.ts (hunk 2 above)
Comment (mmoskal) [reply pxtcompiler/emitter/hexfile.ts]: isAvr() == AVR || AVRVM - this is only for native AVR

**Pull request merged**